core/src/main/java/nl/nn/adapterframework/jdbc/JdbcTransactionalStorage.java

```diff
@@ -706,71 +705,72 @@ private String createResultString(boolean isMessageDifferent){
 			try (Connection conn = getConnection()) {
-				Timestamp receivedDateTime = new Timestamp(receivedDate.getTime());
-				if (messageId.length()>MAXIDLEN) {
-					messageId=messageId.substring(0,MAXIDLEN);
-				}
-				if (correlationId.length()>MAXCIDLEN) {
-					correlationId=correlationId.substring(0,MAXCIDLEN);
-				}
-				if (comments!=null && comments.length()>MAXCOMMENTLEN) {
-					comments=comments.substring(0,MAXCOMMENTLEN);
-				}
-				if (label!=null && label.length()>MAXLABELLEN) {
-					label=label.substring(0,MAXLABELLEN);
-				}
-				result = storeMessageInDatabase(conn, messageId, correlationId, receivedDateTime, comments, label, message);
-				if (result==null) {
-					result=retrieveKey(conn,messageId,correlationId,receivedDateTime);
+				return storeMessage(conn, messageId, correlationId, receivedDate, comments, label, message);
+			} catch (SenderException e) {
+				if (itx != null) {
+					itx.setRollbackOnly();
 				}
-				return result;
-
+				throw e;
 			} catch (Exception e) {
-				if (itx!=null) {
+				if (itx != null) {
 					itx.setRollbackOnly();
 				}
-				throw new SenderException("cannot serialize message",e);
+				throw new SenderException("cannot serialize message", e);
 			}
 		} finally {
-			if (itx!=null) {
+			if (itx != null) {
 				itx.complete();
 			}
 		}
 
 	}
 
-	public String storeMessage(Connection conn, String messageId, String correlationId, Date receivedDate, String comments, String label, S message) throws SenderException {
+	public String storeMessage(@Nonnull Connection conn, @Nonnull String messageId, @Nonnull String correlationId, @Nonnull Date receivedDate, @Nullable String comments, @Nullable String label, @Nonnull S message) throws SenderException {
 		String result;
 		try {
 			Timestamp receivedDateTime = new Timestamp(receivedDate.getTime());
-			if (messageId.length()>MAXIDLEN) {
-				messageId=messageId.substring(0,MAXIDLEN);
+			String storedMessageId;
+			if (messageId.length() > MAXIDLEN) {
+				storedMessageId = messageId.substring(0, MAXIDLEN);
+			} else {
+				storedMessageId = messageId;
 			}
-			if (correlationId.length()>MAXCIDLEN) {
-				correlationId=correlationId.substring(0,MAXCIDLEN);
+			String storedCorrelationId;
+			if (correlationId.length() > MAXCIDLEN) {
+				storedCorrelationId = correlationId.substring(0, MAXCIDLEN);
+			} else {
+				storedCorrelationId = correlationId;
 			}
-			if (comments!=null && comments.length()>MAXCOMMENTLEN) {
-				comments=comments.substring(0,MAXCOMMENTLEN);
+			String storedComments;
+			if (comments != null && comments.length() > MAXCOMMENTLEN) {
+				storedComments = comments.substring(0, MAXCOMMENTLEN);
+			} else {
+				storedComments = comments;
 			}
-			if (label!=null && label.length()>MAXLABELLEN) {
-				label=label.substring(0,MAXLABELLEN);
+			String storedLabel;
+			if (label != null && label.length() > MAXLABELLEN) {
+				storedLabel = label.substring(0, MAXLABELLEN);
+			} else {
+				storedLabel = label;
 			}
```

Comment:
En hier zou je een `truncate` methode voor kunnen maken (in onze StringUtils oid?)

Reply:
Apache StringUtils blijkt er al een te hebben.